python/tests/transform_coords_test.py

```diff
@@ -333,3 +333,10 @@ def to_bd(*, a):
     with pytest.raises(ValueError):
         graph = {('b', 'd'): to_bd, ('b', 'c'): to_bc}
         original.transform_coords(['b'], graph=graph)
+
+
+def test_prioritize_coords_attrs_conflict():
+    original = sc.DataArray(data=a, coords={'a': a}, attrs={'a': -1 * a})
```

Comment:
I actually didn't know you could have a coord and an attr with the same name.
Should we actually prevent this at the core level?

Reply:
I had a discussion with Simon about this but I am not sure if I remember exactly why this is allowed. But one reason is that the attr and coord dicts would have to talk to each other when one inserts variables into them. This would require a pointer to parent or similar which we wanted to avoid.

Reply:
Do we have tests for this at the `DataArray` level? I recall that accessing via `meta` throws in the case that you have same name in both attr and coords, but don't know if we explicitly test for this already.

Reply:
If I understand correctly, this is what this test is doing. The exception is raised because `transform_coord` uses `meta`. Or do you want to know if we have a separate test to check this behaviour of `meta`?

Reply:
> I actually didn't know you could have a coord and an attr with the same name.
> Should we actually prevent this at the core level?

We used to in the old implementation. However, preventing this would actually complicate the code significantly and there is no clear benefit to this.